src/celeritas/optical/surface/SurfaceModelView.hh

```diff
@@ -0,0 +1,110 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/surface/SurfaceModelView.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "celeritas/optical/Types.hh"
+#include "celeritas/phys/SurfacePhysicsMapView.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Optical surface data for a model.
+ *
+ * Wraps common behavior for querying the surface model data for a given
+ * physics surface interface.
+ */
+class SurfaceModelView
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using InternalSurfaceId = SurfacePhysicsMapView::InternalSurfaceId;
```

Comment:
Regarding nomenclature for this ... maybe instead of trying to overload "surface" we can just refer to it as `LocalDataId`? `SubModelId`?  Man there's just too many choices. Since it doesn't have to correspond with a single surface in theory, let's just rename all this to `SubModelId` with the documentation "Internal index of physics data within a single model" (in theory it could be used for surfaces, materials, energy ranges, isotopes, ...) and put it in `celeritas/Types.hh`.